op-proposer/proposer/driver.go

```diff
@@ -269,18 +279,62 @@ func (l *L2OutputSubmitter) FetchL2OOOutput(ctx context.Context) (*eth.OutputRes
 	return output, true, nil
 }
 
-// FetchDGFOutput gets the next output proposal for the DGF.
+// FetchDGFOutput queries the DGF for the latest game and infers whether it is time to make another proposal
+// If necessary, it gets the next output proposal for the DGF, and returns it along with
+// a boolean for whether the proposal should be submitted at all.
 // The passed context is expected to be a lifecycle context. A network timeout
 // context will be derived from it.
-func (l *L2OutputSubmitter) FetchDGFOutput(ctx context.Context) (*eth.OutputResponse, error) {
-	ctx, cancel := context.WithTimeout(ctx, l.Cfg.NetworkTimeout)
+func (l *L2OutputSubmitter) FetchDGFOutput(ctx context.Context) (*eth.OutputResponse, bool, error) {
+	cCtx, cancel := context.WithTimeout(ctx, l.Cfg.NetworkTimeout)
 	defer cancel()
+	callOpts := &bind.CallOpts{
+		From:    l.Txmgr.From(),
+		Context: cCtx,
+	}
 
-	blockNum, err := l.FetchCurrentBlockNumber(ctx)
+	gameCount, err := l.dgfContract.GameCount(callOpts)
 	if err != nil {
-		return nil, err
+		return nil, false, fmt.Errorf("could not query DisputeGameFactory.gameCount(): %w", err)
 	}
-	return l.FetchOutput(ctx, blockNum)
+
+	if gameCount.Sign() == 1 {
+		// If there is at least one game, ensure
+		// enough time has lapsed that we need to make another proposal.
+
+		latestGameIndex := new(big.Int).Sub(gameCount, big.NewInt(1))
+		latestGame, err := l.dgfContract.GameAtIndex(callOpts, latestGameIndex)
+		if err != nil {
+			return nil, false, fmt.Errorf("could not query DisputeGameFactory.GameAtIndex(%d): %w", latestGameIndex, err)
+		}
+
+		timeSinceLastGame := time.Since(time.Unix(int64(latestGame.Timestamp), 0))
+		if timeSinceLastGame <= l.Cfg.ProposalInterval {
+			l.Log.Info("Duration since last game not past proposal interval", "duration", timeSinceLastGame)
+			return nil, false, nil
+		}
+
+		l.Log.Info("Duration since last game past proposal interval, submitting proposal now", "duration", timeSinceLastGame)
+	} else {
+		// If there is not yet any game, wait one interval
+		// and then proceed with making a proposal.
+		// This is to prevent the submitter attempting to
+		// submit too soon after the AnchorStateRegistry is
+		// deployed (mostly only an issue in tests).
+		time.Sleep(l.Cfg.ProposalInterval)
```

Comment:
if we sleep here, and use `cCtx` down below, we should probably reset the `cCtx` to a full network timeout again.